docker/docker.go

```diff
@@ -25,6 +26,14 @@ const (
 )
 
 func main() {
+
+	for j := 1; j < len(os.Args); j++ {
+		if os.Args[j] == "-d" || os.Args[j] == "--daemon" {
+			log.Printf("%s is deprecated. Please use daemon command", os.Args[j])
+			os.Args[j] = "daemon"
```

Comment:
to be backwards compatible, `daemon` should always become the *first* argument, even if `-d` was passed as the x-th argument. The other arguments should be re-ordered to come *after* `daemon`